src/Entities/Player.cpp

```diff
@@ -789,8 +788,7 @@ void cPlayer::SetBedPos(const Vector3i & a_Pos)
 void cPlayer::SetBedPos(const Vector3i & a_Pos, cWorld * a_World)
 {
 	m_LastBedPos = a_Pos;
-	ASSERT(a_World != nullptr);
-	m_SpawnWorldName = a_World->GetName();
+	m_SpawnWorldName = a_World ? a_World->GetName() : m_World->GetName();
```

Comment:
I don't think that's a good idea. Better would be to change `cWorld * a_World` to `cWorld & a_World`. On lua side if a nil is then passed, it will abort the plugin and shows a stack trace in console instead of a crash of the server.

Reply:
Oh well, I actually didn't want this commit to be in this pr. But thanks, I'll have a look into your suggestion!

Reply:
Np. You can create a own pr for that.